packages/web-api/src/controllers/scan-report-controller.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { GuidGenerator, ServiceConfiguration } from 'common';
+import { inject, injectable } from 'inversify';
+import { Logger } from 'logger';
+import { ApiController, PageScanRunReportService } from 'service-library';
+import { Stream } from 'stream';
+
+@injectable()
+export class ScanReportController extends ApiController {
+    public readonly apiVersion = '1.0';
+    public readonly apiName = 'web-api-get-report';
+
+    public constructor(
+        @inject(PageScanRunReportService) private readonly pageScanRunReportService: PageScanRunReportService,
+        @inject(GuidGenerator) protected readonly guidGenerator: GuidGenerator,
+        @inject(ServiceConfiguration) protected readonly serviceConfig: ServiceConfiguration,
+        @inject(Logger) private readonly logger: Logger,
+    ) {
+        super();
+    }
+
+    public async handleRequest(): Promise<void> {
+        const reportId = <string>this.context.bindingData.reportId;
+        if (!this.guidGenerator.isValidV6Guid(reportId)) {
+            this.context.res = {
+                status: 422,
+                body: `Invalid report id: ${reportId}.`,
+            };
+
+            return;
+        }
+
+        const blobContentDownloadResponse = await this.pageScanRunReportService.readSarifReport(reportId);
+
+        if (blobContentDownloadResponse.notFound === true) {
+            this.context.res = {
+                status: 404,
+            };
+
+            return;
+        }
+
+        const stream = new Stream.PassThrough();
+        blobContentDownloadResponse.content.pipe(stream);
```

Comment:
can't we directly pass the stream returned by blobContentDownloadResponse.content?

Reply:
Or is this the pattern when dealing with rest api response?

Reply:
With [other frameworks](https://dev.azure.com/mseng/_git/AzureDevOps?path=%2FKeros%2FService%2FServer%2FUtility%2FHttpResponseMessageCreator.cs&version=GBmaster) it is possible to send a stream. Not sure whether we can do it with typescript. We can also punt this to a bug.

Reply:
I passed content as body directly initially but the client never get response.